openpype/hosts/houdini/plugins/publish/validate_unreal_staticmesh_naming.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+"""Validator for correct naming of Static Meshes."""
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from openpype.pipeline.publish import ValidateContentsOrder
+
+from openpype.hosts.houdini.api.action import SelectInvalidAction
+from openpype.hosts.houdini.api.lib import get_output_children
+
+import hou
+
+
+class ValidateUnrealStaticMeshName(pyblish.api.InstancePlugin,
+                                   OptionalPyblishPluginMixin):
+    """Validate name of Unreal Static Mesh.
+
+    This validator checks if output node name has a collision prefix:
+            - UBX
+            - UCP
+            - USP
+            - UCX
+
+    This validator also checks if subset name is correct
+            - {static mesh prefix}_{Asset-Name}{Variant}.
+
+    """
+
+    families = ["staticMesh"]
+    hosts = ["houdini"]
+    label = "Unreal Static Mesh Name (FBX)"
+    order = ValidateContentsOrder + 0.1
+    actions = [SelectInvalidAction]
+
+    optional = True
+
+    @classmethod
+    def apply_settings(cls, project_settings, system_settings):
+        settings = (
+            project_settings["houdini"]["create"]["CreateUnrealStaticMesh"]
+        )
+        cls.collision_prefixes = settings["collision_prefixes"]
+        cls.static_mesh_prefix = settings["static_mesh_prefix"]
```

Comment:
This is lacking the reading of settings for this plug-ins `optional`, `active` and `enabled` state from `houdini/publish/ValidateUnrealStaticMeshName`. Since you've overridden the function you need to apply it yourself as far as I know.